__tests__/app.test.js

```diff
@@ -167,5 +168,53 @@ describe("POST", () => {
                 expect(comment.article_id).toBe(1)
             })
         })
+        test("400 - when the body does only contains username", () => {
+            return request(app)
+            .post("/api/articles/1/comments")
+            .send({
+                username: "butter_bridge"
+            })
+            .expect(400)
+            .then(({body: {msg}}) => {
+                expect(msg).toBe("Missing data for post request, please ensure a body with keys 'username' and 'body'")
+            })
+        })
+        test("400 - when the body does only contains body", () => {
+            return request(app)
+            .post("/api/articles/1/comments")
+            .send({
+                body: "This movie deserves 5 gigantic stars"
+            })
+            .expect(400)
+            .then(({body: {msg}}) => {
+                expect(msg).toBe("Missing data for post request, please ensure a body with keys 'username' and 'body'")
+            })
+        })
+        test("404 - when provided a non existent article ID", () => {
+            return request(app)
+            .post("/api/articles/999/comments")
+            .send({
+                username: "butter_bridge",
+                body: "This movie deserves 5 gigantic stars",
+                votes: 5
+            })
+            .expect(404)
+            .then(({body: {msg}}) => {
+                expect(msg).toBe("Key (article_id)=(999) is not present in table \"articles\".")
+            })
+        })
+        test("400 - when provided an invalid article ID", () => {
+            return request(app)
+            .post("/api/articles/dragons/comments")
+            .send({
+                username: "butter_bridge",
+                body: "This movie deserves 5 gigantic stars",
+                votes: 5
+            })
+            .expect(400)
+            .then(({body: {msg}}) => {
+                expect(msg).toBe("Invalid datatype of input")
+            })
+        })
```

Comment:
really good error testing - one more you can add is that the username must exist in the database. see if SQL will deal with that for you before trying a check exists from the lecture